ros-build-essential/README.md

```diff
@@ -0,0 +1,20 @@
+# ROS Build Essential
+
+`ros-build-essential` includes packages that are expected to be present when building any ROS packages without an explicit dependency.
+It should include the base toolchain for any operating system (such as `build-essential`) on Debian as well as Make and a C++ compiler if not already part of that toolchain.
+
+### Platform-specific build toolchains
+
+This variant is meant primarily for internal use by the ROS Infrastructure team to set up the environment in which packages will be built on the ROS build farm.
+Since this package is used by the more user-facing ros-dev-tools variant as well, it can be useful to define the variant for platforms that are not currently enabled, or potentially even supported, by the build farm infrastructure.
+When selecting packages to include in this variant, the platform's own build toolchain, including any utilities used in the package creation process, should be used as a starting point.
+For example the `build-essential` package in Debian, `base-devel` in Arch Linux, or the `Development Tools` group in RHEL.
+
+If the default build toolchain for a distribution does not include GNU Make and a C++ compiler, those should be included as well.
```

Comment:
@cottsay this is my attempt to capture the outcome of our discussion yesterday about how to handle this across multiple platforms. I'd appreciate your careful review here to make sure that the above is clear, complete, and matches your recollection.

Reply:
> ... or the `Development Tools` group in RHEL.

I'm not sure this is a very good example to give. Fedora's story here is a lot cleaner.

```suggestion
### Platform-specific build toolchains

This variant is meant primarily for internal use by the ROS Infrastructure team to set up the environment in which packages will be built on the ROS build farm.
Since this package is used by the more user-facing ros-dev-tools variant as well, it can be useful to define the variant for platforms that are not currently enabled, or potentially even supported, by the build farm infrastructure.
When selecting packages to include in this variant, the platform's own build toolchain, including any utilities used in the package creation process, should be used as a starting point.
For example the `build-essential` package in Debian, `base-devel` in Arch Linux, or the `Buildsystem building group` in Fedora.

If the default build toolchain for a distribution does not include GNU Make and a C++ compiler, those should be included as well.
```

Reply:
Hmm, part of the reason I wanted to include an Enterprise Linux example is that it's our other best supported platform at the moment. Does it make sense to instead call out the `mock` template group for the rpm platforms as that's what we're actually implementing?

Reply:
That would be better, and by contrast factually correct, though still not as "clean" as Fedora's and Debian's stories.

Reply:
I updated our RHEL example and included the Fedora example too. If we get contributions it may make sense to break this out into full on list of platforms but I'm not ready to do so just yet.

[d41b38c](https://github.com/ros-infrastructure/infra-variants/pull/1/commits/d41b38c2f8946faf1c1237e59c25313fa91f4b51)